tests/management/workspace/test_view.py

```diff
@@ -0,0 +1,340 @@
+        self.assertNotEquals(data.get("uuid"), "")
+        self.assertIsNotNone(data.get("uuid"))
+        self.assertNotEquals(data.get("created"), "")
+        self.assertNotEquals(data.get("modified"), "")
+        self.assertEquals(data.get("description"), "Other description")
+
+    def test_create_workspace_empty_body(self):
+        """Test for creating a workspace."""
+        workspace = {}
+
+        url = reverse("workspace-list")
+        client = APIClient()
+        response = client.post(url, workspace, format="json", **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        error = response.data.get("errors")[0]
+        self.assertIsNotNone(error.get("detail"))
+        self.assertEqual(error.get("detail"), "This field is required.")
+        self.assertEqual(error.get("source"), "name")
+        self.assertEqual(error.get("status"), "400")
+
+    def test_create_workspace_unauthorized(self):
+        """Test for creating a workspace."""
+        workspace = {}
+
+        request_context = self._create_request_context(self.customer_data, self.user_data, is_org_admin=False)
+
+        request = request_context["request"]
+        headers = request.META
+
+        url = reverse("workspace-list")
+        client = APIClient()
+        response = client.post(url, workspace, format="json", **headers)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        error = response.data.get("errors")[0]
+
+        self.assertEqual(error.get("detail"), "You do not have permission to perform this action.")
+        self.assertEqual(error.get("source"), "detail")
+        self.assertEqual(error.get("status"), "403")
+
+    def test_duplicate_create_workspace(self):
+        """Test that creating a duplicate workspace is not allowed."""
+        name = self.init_workspace_attributes["name"]
+        test_data = {"name": name}
+
+        url = reverse("workspace-list")
+        client = APIClient()
+        response = client.post(url, test_data, format="json", **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        error = response.data.get("errors")[0]
+        self.assertIsNotNone(error.get("detail"))
+        self.assertEqual(error.get("detail"), "Workspace already exist in tenant")
+        self.assertEqual(error.get("source"), "workspace")
+        self.assertEqual(error.get("status"), "400")
+
+    def test_update_workspace(self):
+        """Test for updating a workspace."""
+        workspace_data = {
+            "name": "New Workspace",
+            "description": "New Workspace - description",
+            "tenant_id": self.tenant.id,
+        }
+
+        workspace = Workspace.objects.create(**workspace_data)
+
+        url = reverse("workspace-detail", kwargs={"uuid": workspace.uuid})
+        client = APIClient()
+
+        workspace_data["name"] = "Updated name"
```

Comment:
I think the validation currently requires that the `name` be updated in a `PUT`. For instance, I've tried the following example requests:

Given the following workspace record:
```json
{"name": "foo", "description": "foo workspace", "uuid": "1234"}
```

- `PUT` request `/api/rbac/v2/workspaces/1234/` with payload `{"name": "foo", "description": "UPDATED"}` I get back `{"errors":[{"detail":"Workspace already exist in tenant","source":"workspace","status":"400"}]}` -- it's checking uniqueness when I don't want/need to update the name, but only want to update description here
- `PUT` request `/api/rbac/v2/workspaces/1234/` with payload `{"description": "UPDATED"}` I get back `{"errors":[{"detail":"This field is required.","source":"name","status":"400"}]}` -- should actually be a `PATCH` to update only certain fields, so this one probably makes sense.

I'd expect

Reply:
😄